src/System.IO.Abstractions.TestingHelpers/MockFileStream.cs

```diff
@@ -71,6 +74,20 @@ public class MockFileStream : MemoryStream
         /// <inheritdoc />
         public override bool CanWrite => access.HasFlag(FileAccess.Write);
 
+        /// <inheritdoc />
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            fileData.LastAccessTime = DateTime.Now;
+            return base.Read(buffer, offset, count);
+        }
+
+        /// <inheritdoc />
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            fileData.LastWriteTime = fileData.LastAccessTime = DateTime.Now;
+            base.Write(buffer, offset, count);
+        }
+
```

Comment:
I have to admit that right now I do not know if the real FileStream implementation is updating the LastAccessTime or
the LastWriteTime on any Read/Write. I could also imagine that it kind of bulk reads / writes and only on `Flush` it will
do the updates.

But as I said, I have not checked. Some tests or comments or links to the documentation from you would help to clarify that change.

Reply:
In fact, I too have not examined the `FileStream` implementation in that regard.  However, I have noticed that the timestamps are not always updated immediately after the respective operation, but at the latest when the OS has processed the flush.

In contrast, with the `MockFileStream` implementation, I made the assumption for simplicity that the IO operations are executed immediately and the data is persisted instantaneously.

Reply:
Thanks for checking! I agree that we can go for the simple solution here. I think there is no exact guarantee of when these timestamps are set even in the real FS (depends on OS).